File: PalindromeNumber.java

public class PalindromeNumber {
    public static boolean isPalindrome(int x) {
        if (0 <= x && x <= 9) {
            return true;
        }   
        if (x < 0 || x % 10 == 0) {
            return false;
        }   
        int y = 0;
        while (x > y) {
            y = y * 10 + x % 10; 
            x = x / 10;
        }   
        return (x == y || y / 10 == x);
    }
    public static void main(String[] args) {
        System.out.println(isPalindrome(2345432));
        System.out.println(isPalindrome(-23));
    }
}

File: py_solution/palindrome_number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if 0 <= x and x <= 9:
            return True
        if x < 0 or x % 10 == 0:
            return False
        y = 0
        while x > y:
            y = y * 10 + x % 10
            x = int(x / 10)
        return x == y or int(y / 10) == x


if __name__ == "__main__":
    solution = Solution()
    assert solution.isPalindrome(2345432) == True
    assert solution.isPalindrome(-23) == False

